File: frontend/scripts/inspect-selection-xlsx.py
"""선발 명단 xlsx 파일 구조 + 데이터 추출."""
import sys, os, json
import openpyxl

path = r"C:\Users\USER\Downloads\1. 2026년 AI챔피언 및 공공 AI역량 트랙 교육 (6월2차) 선발 명단_수정 (1).xlsx"
wb = openpyxl.load_workbook(path, data_only=True)

print("=== 시트 목록 ===")
for sn in wb.sheetnames:
    ws = wb[sn]
    print(f"  '{sn}': {ws.max_row} rows × {ws.max_column} cols")

# 각 시트의 첫 2행 (헤더 + 첫 데이터) 표시
for sn in wb.sheetnames:
    ws = wb[sn]
    print(f"\n=== '{sn}' 헤더 ===")
    rows = list(ws.iter_rows(min_row=1, max_row=3, values_only=True))
    for i, r in enumerate(rows):
        print(f"  row {i}: {' | '.join(str(c) if c is not None else '' for c in r[:15])}{' ...' if len(r) > 15 else ''}")

File: frontend/scripts/inspect-selection-xlsx.ts
import * as XLSX from 'xlsx';
import path from 'path';

const file = process.argv[2] ?? path.join('C:', 'Users', 'USER', 'Downloads', '1. 2026년 AI챔피언 및 공공 AI역량 트랙 교육 (6월2차) 선발 명단.xlsx');
console.log('파일:', file);

const wb = XLSX.readFile(file);
console.log('시트:', wb.SheetNames);

for (const sn of wb.SheetNames) {
  const ws = wb.Sheets[sn];
  const rows = XLSX.utils.sheet_to_json<unknown[]>(ws, { header: 1, defval: '' });
  console.log(`\n=== ${sn} === 총 ${rows.length}행`);
  for (let i = 0; i < Math.min(8, rows.length); i++) {
    const r = rows[i] as unknown[];
    console.log(`  row${i}:`, JSON.stringify(r).slice(0, 300));
  }
}
